src/bp-templates/bp-nouveau/includes/activity/ajax.php

```diff
@@ -1127,3 +1140,68 @@ function bb_nouveau_ajax_activity_update_pinned_post() {
 		wp_send_json_error( $response );
 	}
 }
+
+/**
+ * Update close activity comments.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @return void
+ */
+function bb_nouveau_ajax_activity_update_close_comments() {
+	$response = array(
+		'feedback' => esc_html__( 'There was a problem marking this operation. Please try again.', 'buddyboss' ),
+	);
+
+	if ( ! bp_is_post_request() ) {
+		wp_send_json_error( $response );
+	}
+
+	if ( ! is_user_logged_in() ) {
+		wp_send_json_error( $response );
+	}
+
+	// Nonce check!
+	if ( empty( $_POST['nonce'] ) || ! wp_verify_nonce( $_POST['nonce'], 'bp_nouveau_activity' ) ) {
+		wp_send_json_error( $response );
+	}
+
+	if ( empty( $_POST['close_comments_action'] ) ) {
+		wp_send_json_error( $response );
+	}
+
+	if ( empty( $_POST['id'] ) ) {
+		wp_send_json_error( $response );
+	}
+
+	if ( ! in_array( $_POST['close_comments_action'], array( 'close_comments', 'unclose_comments' ), true ) ) {
+		wp_send_json_error( $response );
+	}
```

Comment:
@yudhisthirnahar You are generating the identical error message for each incorrect condition. Consequently, evaluate all these conditions collectively and return the error.